crates/gen/src/types/struct.rs

```diff
@@ -68,6 +70,35 @@ impl Struct {
                     self as *mut Self::Abi
                 }
             }
+            #extensions
+        }
+    }
+
+    fn extension_tokens(&self) -> TokenStream {
+        if self.name.name == "TimeSpan" && self.name.namespace == "Windows.Foundation" {
```

Comment:
Why do this in generated code instead of in the WinRT crate proper? I would imagine that we want to put as much into "normal" code as we can and only generate what we have to.